### PR TITLE
Milestone 10 : Views specs :test_tube:

### PR DESCRIPTION
**Pull request by HAMMAS-SALEEM**
# Note for Reviewers:person_curly_hair: 
### In this branch, I (@HAMMAS-SALEEM) have worked on the following things :+1: 

:heavy_check_mark: Installed Capybara
:heavy_check_mark: Added Tests for User
:heavy_check_mark: Added Tests for Post
:heavy_check_mark: Added Tests for Login page
:heavy_check_mark: Updated Tests for Controller
:heavy_check_mark: Updated Tests for Models
:heavy_check_mark: Followed Linters
:heavy_check_mark: Followed Ruby best practices
:heavy_check_mark: Documented work in a professional way

### CODE REVIEW

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @HAMMAS-SALEEM,** 

### **Good job so far!**
You still need to work on some issues to go to the next project, but you are almost there!

## **Highlights**

- Nice PR ✅.
- Good readme file ✅.
- Linters are good ✅.


## Required Changes ♻️
_Check the comments under the review._

- Hello @HAMMAS-SALEEM excellent job on adding tests suits to your rails app however some of your tests are failing and others are still pending so please check that all tests pass before re-submitting again. 

Capture d’écran 2022-05-27 125957.

Capture d’écran 2022-05-27 130830



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HAMMAS-SALEEM** [comment]
Thanks for giving me a review. I have refactored the tests. Now all tests are passing. Pending tests were just empty files.
tests

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @HAMMAS-SALEEM ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.




### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

- [ ] As the last code reviewer said please fix the errors

image


___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Comment by HAMMAS-SALEEM** [comment]
# Note for Reviewers:person_curly_hair: 
### **These tests are not passing because you guys are using the old ruby version. Please install ruby version 3.1.1 or 3.1.2. All the tests will pass on these versions.**

**Comment by dasileker** [comment]
> # Note for Reviewers:person_curly_hair: 
> ### **These tests are not passing because you guys are using the old ruby version. Please install ruby version 3.1.1 or 3.1.2. All the tests will pass on these versions.**

Hello @HAMMAS-SALEEM this is @dasileker

You are right, for me I use rvm  for each student I change my ruby version, so in your case mine you have 24 tests passing and 33 not passing.

File: spec/features/post_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      expect(page).to have_content 'Ani'
+      expect(page).to have_content 'Vahan'
+    end
+
+    it 'shows the comment of the commentor' do
+      expect(page).to have_content 'My first comment'
+      expect(page).to have_content 'My second comment'
+    end
+  end
+end
```
Comment (Gambit142) [inline spec/features/post_show_spec.rb:56]: - [ ] When I run your features test folder, 8 examples fail. The reason for the failure is the syntax error on line 42 of your `./app/views/posts/show.html.erb` file. Your comment key is not taking any value. Kindly fix it and all your tests will pass. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @HAMMAS-SALEEM 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_show_spec.rb (hunk 1 above)
Comment (HAMMAS-SALEEM) [reply spec/features/post_show_spec.rb:56]: Hey Bro

How are you?

You have given me a review. Thanks for the review.

Everything test is passed on my machine.

Every test is passing. I have checked on 2 other partners' laptops and every test passed.

I guess you are using the ruby old version. Please install the latest version of ruby. Then every test will pass.

**Comment by HAMMAS-SALEEM** [comment]
# Note for Reviewers:person_curly_hair: 
### **In my previous submissions, All tests are not passing because you guys are using the old ruby version. One of my reviewers was arguing with me that it should work. Please install ruby version 3.1.1 or 3.1.2. All the tests will pass on these versions. This is my extra review. Please don't waste my reviews.**

### Please install the latest version of ruby at least 3.1.1. I have tried it. I have checked it on multiple computers with multiple versions. I know it works on the latest version at least 3.1.1. Please don't waste my review by requesting changes again and again. Otherwise, I have to report student success.

File: spec/features/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+RSpec.describe 'Posts show page', type: :feature do
+  before(:each) do
+    @user = User.create(Name: 'Ani', Photo: 'img.jpg', Bio: 'Adviser', email: 'test@email.com',
+                        password: 'password', confirmed_at: Time.now, Posts_Counter: 0)
+    @user.confirm
+    @user1 = User.create(Name: 'Vahan', Photo: 'img.jpg', Bio: 'Developer', email: 'test1@email.com',
+                         password: 'password', confirmed_at: Time.now, Posts_Counter: 0)
+    @user1.confirm
+    @post = Post.create(Title: 'My title', Text: 'My text', CommentsCounter: 0, LikesCounter: 0, user_id: @user.id)
```
Comment (uwadonat) [inline spec/features/post_show_spec.rb:11]: - [x] All your tests are passing successfully.
Screenshot from 2022-05-28 03-44-22

**Review by uwadonat — APPROVED** [review]
Hi **@[HAMMAS-SALEEM](https://github.com/HAMMAS-SALEEM),** 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- Your commit messages are very descriptive ✔️
- You implemented well JavaScript best practices ✔️
- No linter errors ✔️
appr3


# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/features/post_show_spec.rb (hunk 2 above)
Comment (HAMMAS-SALEEM) [reply spec/features/post_show_spec.rb:11]: Thanks, bro for giving me a good review.

**Pull request merged**